### PR TITLE
[1LP][RFR] Fix cloud event listener

### PR DESCRIPTION
**Pull request by ohiliazo**
PURPOSE
=================

__Fixing__ quickstart script because it was failing when the OS version wasn't recognized.
__Fixing__ test_cloud_events because now in 5.9 the response is dict instead of yaml stream.
__Minor changes__ to REST event listener get_next_portion func.

Shriver:
{{ pytest: cfme/tests/cloud/test_cloud_events.py --long-running --use-provider azure}}

### CODE REVIEW

File: cfme/tests/cloud/test_cloud_events.py [hunk 1]
```diff
@@ -26,7 +25,7 @@ def test_manage_nsg_group(provider, setup_provider, register_event):
     # we need to check raw data by regexps, since many azure events aren't parsed by CFME yet
 
     def add_cmp(_, y):
-        data = yaml.load(y)
+        data = yaml.load(y) if provider.appliance.version < '5.9' else y
```
Comment (jamesooden) [inline cfme/tests/cloud/test_cloud_events.py]: Add a comment explaining that in 5.9 y is a dict not a yaml stream.

File: cfme/tests/cloud/test_cloud_events.py [hunk 2]
```diff
@@ -40,7 +39,7 @@ def add_cmp(_, y):
                    event_type='networkSecurityGroups_write_EndRequest')
 
     def rm_cmp(_, y):
-        data = yaml.load(y)
+        data = yaml.load(y) if provider.appliance.version < '5.9' else y
```
Comment (jamesooden) [inline cfme/tests/cloud/test_cloud_events.py]: Add a comment explaining that in 5.9 y is a dict not a yaml stream.

File: cfme/tests/cloud/test_cloud_events.py [hunk 3]
```diff
@@ -71,8 +70,9 @@ def test_vm_capture(request, provider, setup_provider, register_event):
     request.addfinalizer(vm.delete_from_provider)
 
     def cmp_function(_, y):
-        data = yaml.load(y)
+        data = yaml.load(y) if provider.appliance.version < '5.9' else y
```
Comment (jamesooden) [inline cfme/tests/cloud/test_cloud_events.py]: Add a comment explaining that in 5.9 y is a dict not a yaml stream.

**Comment by mshriver** [comment]
No tests collected for PRT run 22498, false positive.

File: cfme/tests/cloud/test_cloud_events.py [hunk 4]
```diff
@@ -40,7 +41,9 @@ def add_cmp(_, y):
                    event_type='networkSecurityGroups_write_EndRequest')
 
     def rm_cmp(_, y):
-        data = yaml.load(y)
+        # In 5.9 version `y` is a dict, not a yaml stream.
+        data = yaml.load(y) if provider.appliance.version < '5.9' else y
```
Comment (RonnyPfannschmidt) [inline cfme/tests/cloud/test_cloud_events.py]: **question** should we move the version dependent de-serialize to the framework side

the usage here seems limited, but i wonder if there are more call sites that need the behaviour

**Pull request merged**